Konverzni funkce

Prokopl opened the pull request:
Vytvorena funkce convert, ktera rozhodne, co s tokenem.
Vytvorena funcke keywordFilter na filtrovani klicovych slov.
Otestovano na zkusebnich datech - ruzne formaty u double bez problemu.

Prokopl committed:
Konverzni funkce

Vytvorena funkce convert, ktera rozhodne, co s tokenem.
Vytvorena funcke keywordFilter na filtrovani klicovych slov.
Otestovano na zkusebnich datech - ruzne formaty u double bez problemu.

MWarCZ commented:
1) Funkce keywordFilter je nefunkční neboť v TokenTypes jsou opisy názvů tokenu tj. TK_AND, TK_IF.
Takže ty porovnáváš TK_AND a and takže se typ tokenu nikdy nezmění.
2) Pojmenováváme funkce tak že začínají velkým písmenem ConvertToken, KeywordFilter, ...

Prokopl committed:
Konverzni funkce

Funkce prejmenovane, aby zacinaly velkym pismenem.
Opravena funkce KeywordFilter, aby porovnala retezec s tokenem.

MWarCZ committed:
Opravena funkce KeywordFilter

Funkce KeywordFilter nefungovala. 
1) V porovnávaném pomocný string nebyl prázdný a proto nedocházelo ke splnění podmínky.
2) Docházelo k přístupu mimo alokovanou paměť.
